redfish-core/lib/cable.hpp

```diff
@@ -73,6 +126,175 @@ inline void
+                    }
+                    else
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "in Downstream Resources " << fullPath
+                            << " isn't found in chassis assembly list";
+                    }
+                }
+            });
+    };
+
+    std::string upstreamConnector = cableObjectPath + "/upstream_connector";
+    linkAssociatedCable(
+        asyncResp, upstreamConnector,
+        [asyncResp](const std::vector<std::string>& value) {
+            nlohmann::json& linkArray =
+                asyncResp->res.jsonValue["Links"]["UpstreamPorts"];
+            linkArray = nlohmann::json::array();
+
+            for (const auto& fullPath : value)
+            {
+                sdbusplus::message::object_path path(fullPath);
+                std::string leaf = path.filename();
+                if (leaf.empty())
+                {
+                    continue;
+                }
+                std::string endpointLeaf = path.parent_path().filename();
+                if (endpointLeaf.empty())
+                {
+                    continue;
+                }
+
+                //  insert/create link using endpointLeaf
+                endpointLeaf += "/Ports/";
+                endpointLeaf += leaf;
+                linkArray.push_back(
+                    {{"@odata.id",
+                      "/redfish/v1/Systems/system/FabricAdapters/" +
+                          endpointLeaf}});
+            }
+        });
+
+    redfish::chassis_utils::getChassisAssembly(asyncResp, "chassis",
+                                               std::move(chassisAssociation));
+
+    std::string downstreamConnector = cableObjectPath + "/downstream_connector";
+    linkAssociatedCable(
+        asyncResp, downstreamConnector,
+        [asyncResp](const std::vector<std::string>& value) {
+            nlohmann::json& linkArray =
+                asyncResp->res.jsonValue["Links"]["DownstreamPorts"];
+            linkArray = nlohmann::json::array();
+
+            for (const auto& fullPath : value)
+            {
+                sdbusplus::message::object_path path(fullPath);
+                std::string leaf = path.filename();
+                if (leaf.empty())
+                {
+                    continue;
+                }
+                std::string endpointLeaf = path.parent_path().filename();
+                if (endpointLeaf.empty())
+                {
+                    continue;
+                }
+
+                //  insert/create link suing endpointLeaf
+                endpointLeaf += "/Ports/";
+                endpointLeaf += leaf;
+                linkArray.push_back(
+                    {{"@odata.id",
+                      "/redfish/v1/Systems/system/FabricAdapters/" +
+                          endpointLeaf}});
+            }
+        });
+
+    std::string upstreamChassis = cableObjectPath + "/upstream_chassis";
```

Comment:
I'm not sure about this "upstream_chassis" as I couldn't find entries in the rainier association file (ibm,rainier-2u_associations.json). Is this something new that is yet to be merged?